### PR TITLE
Update mower behavior skip path option added

### PR DESCRIPTION
**Pull request by midevil**

### CODE REVIEW

File: src/mower_logic/src/mower_logic/behaviors/MowingBehavior.cpp [hunk 1]
```diff
@@ -363,6 +364,13 @@ bool MowingBehavior::execute_mowing_plan() {
                         skip_area = false;
                         return true;
                     }
+                    if(skip_path) 
+                    {
+                        currentMowingPaths.erase(currentMowingPaths.begin());
+                        skip_path=false;
+                        return true;
```
Comment (olliewalsh) [inline src/mower_logic/src/mower_logic/behaviors/MowingBehavior.cpp:371]: should return false

**Review by olliewalsh — CHANGES_REQUESTED** [review]

File: src/mower_logic/src/mower_logic/behaviors/MowingBehavior.cpp (hunk 1 above)
Comment (midevil) [reply src/mower_logic/src/mower_logic/behaviors/MowingBehavior.cpp:371]: Are you sure? It also returns true when doing a skip area?

Comment (olliewalsh) [reply src/mower_logic/src/mower_logic/behaviors/MowingBehavior.cpp:371]: yes, look at the end of this method - true means all paths complete. Can either return false here or return `currentMowingPaths.empty();`

Comment (midevil) [reply src/mower_logic/src/mower_logic/behaviors/MowingBehavior.cpp]: check, fixed it

File: .github/workflows/build-image.yaml [hunk 2]
```diff
@@ -7,11 +7,13 @@ on:
     branches:
       - 'main'
       - 'releases/**'
+      - 'startinzone'
     tags:
       - 'v*.*.*'
   pull_request:
     branches:
       - 'main'
+      - 'startinzone'
```
Comment (olliewalsh) [inline .github/workflows/build-image.yaml:16]: Need to remove

File: src/mower_logic/cfg/MowerLogic.cfg [hunk 3]
```diff
@@ -29,5 +29,5 @@ gen.add("add_fake_obstacle", bool_t, 0, "True to add a fake obstacle to hopefull
 gen.add("ignore_gps_errors", bool_t, 0, "True to ignore gps errors. USE ONLY FOR SIMULATION!", False)
 gen.add("max_first_point_attempts", int_t, 0, "Maximum attempts to reach the first point of the mow path before trimming", 3, 1, 10)
 gen.add("max_first_point_trim_attempts", int_t, 0, "After <max_first_point_attempts> we start to trim the path beginning this often", 3, 1, 10)
-
+gen.add("clear_path_on_start", bool_t, 0, "True to reset current mowing path on start, useful to start in given area", False)
```
Comment (olliewalsh) [inline src/mower_logic/cfg/MowerLogic.cfg:32]: Not used

File: .github/workflows/build-image.yaml [hunk 4]
```diff
@@ -7,11 +7,13 @@ on:
     branches:
       - 'main'
       - 'releases/**'
+      - 'startinzone'
```
[the remark this answers is not in the document]
Comment (ClemensElflein) [reply .github/workflows/build-image.yaml:10]: done, thank you

File: .github/workflows/build-image.yaml (hunk 2 above)
Comment (ClemensElflein) [reply .github/workflows/build-image.yaml:16]: done, thank you

File: src/mower_logic/cfg/MowerLogic.cfg (hunk 3 above)
Comment (ClemensElflein) [reply src/mower_logic/cfg/MowerLogic.cfg:32]: removed

**Review by ClemensElflein — APPROVED** [review]
LGTM, thanks for the PR

**Pull request merged**